Apps/Common/SIMSolver.h

```diff
@@ -195,12 +196,36 @@ template<class T1> class SIMSolver : public SIMadmin
     return restartStep;
   }
 
+  //! \brief Handles application data output.
```

Comment:
aha, i see now, it sets the internal pointer but still returns a reference to the world. that doesn't seem right.